### ISSUE TITLE
pixiedust is unable to share python value with scala

### ISSUE THREAD

**Issue by snowch** [ibm-cds-labs/pixiedust#18]
See:

<img width="1508" alt="screen shot 2016-10-31 at 01 49 51" src="https://cloud.githubusercontent.com/assets/147713/19842036/63b6e94a-9f0c-11e6-93ed-e25dd07f2f2d.png">

The full stacktrace:

```
Py4JJavaErrorTraceback (most recent call last)
<ipython-input-19-1128de37cfc4> in <module>()
----> 1 get_ipython().run_cell_magic(u'scala', u'', u'\nprintln(bootstrap_servers_string)\n\nimport org.apache.spark.streaming.Duration\nimport org.apache.spark.streaming.Seconds\nimport org.apache.spark.streaming.StreamingContext\nimport com.ibm.cds.spark.samples.config.MessageHubConfig\nimport com.ibm.cds.spark.samples.dstream.KafkaStreaming.KafkaStreamingContextAdapter\nimport org.apache.kafka.common.serialization.Deserializer\nimport org.apache.kafka.common.serialization.StringDeserializer\n\nval kafkaProps = new MessageHubConfig\n\nkafkaProps.setConfig("bootstrap.servers", bootstrap_servers_string.toString())\nkafkaProps.setConfig("kafka.user.name", sasl_plain_username.toString())\nkafkaProps.setConfig("kafka.user.password", sasl_plain_password.toString())\nkafkaProps.setConfig("kafka.topic", messagehub_topic_name.toString())\nkafkaProps.setConfig("api_key", api_key.toString())\nkafkaProps.setConfig("kafka_rest_url", kafka_rest_url.toString())\n\nkafkaProps.createConfiguration()\n\nval ssc = new StreamingContext( sc, Seconds(2) )\n\nval stream = ssc.createKafkaStream[String, String, StringDeserializer, StringDeserializer](\n                     kafkaProps,\n                     List(kafkaProps.getConfig("kafka.topic"))\n                     );\n\nstream.print()\nssc.start()\nssc.awaitTermination()')

/usr/local/src/bluemix_jupyter_bundle.v22/notebook/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc in run_cell_magic(self, magic_name, line, cell)
   2118             magic_arg_s = self.var_expand(line, stack_depth)
   2119             with self.builtin_trap:
-> 2120                 result = fn(magic_arg_s, cell)
   2121             return result
   2122 

<decorator-gen-125> in scala(self, line, cell)

/usr/local/src/bluemix_jupyter_bundle.v22/notebook/lib/python2.7/site-packages/IPython/core/magic.pyc in <lambda>(f, *a, **k)
    191     # but it's overkill for just that one bit of state.
    192     def magic_deco(arg):
--> 193         call = lambda f, *a, **k: f(*a, **k)
    194 
    195         if callable(arg):

/gpfs/fs01/user/s85d-88ebffb000cc3e-39ca506ba762/.local/lib/python2.7/site-packages/pixiedust/utils/scalaBridge.pyc in scala(self, line, cell)
    166                 runnerObject.callMethod("set" + key[0].upper() + key[1:], val["initValue"])
    167 
--> 168         varMap = runnerObject.callMethod("runCell")
    169 
    170         #capture the return vars and update the interactive shell

/gpfs/fs01/user/s85d-88ebffb000cc3e-39ca506ba762/.local/lib/python2.7/site-packages/pixiedust/utils/javaBridge.pyc in callMethod(self, methodName, *args)
    145                             break;
    146                 if match:
--> 147                     return m.invoke(self.jHandle, jMethodArgs)
    148 
    149         raise ValueError("Method {0} that matches the given arguments not found".format(methodName) )

/usr/local/src/spark160master/spark/python/lib/py4j-0.9-src.zip/py4j/java_gateway.py in __call__(self, *args)
    811         answer = self.gateway_client.send_command(command)
    812         return_value = get_return_value(
--> 813             answer, self.gateway_client, self.target_id, self.name)
    814 
    815         for temp_arg in temp_args:

/usr/local/src/spark160master/spark/python/pyspark/sql/utils.py in deco(*a, **kw)
     43     def deco(*a, **kw):
     44         try:
---> 45             return f(*a, **kw)
     46         except py4j.protocol.Py4JJavaError as e:
     47             s = e.java_exception.toString()

/usr/local/src/spark160master/spark/python/lib/py4j-0.9-src.zip/py4j/protocol.py in get_return_value(answer, gateway_client, target_id, name)
    306                 raise Py4JJavaError(
    307                     "An error occurred while calling {0}{1}{2}.\n".
--> 308                     format(target_id, ".", name), value)
    309             else:
    310                 raise Py4JError(

Py4JJavaError: An error occurred while calling o513.invoke.
: java.lang.NullPointerException
    at com.ibm.pixiedust.PixiedustScalaRun$.runCell(pixiedustRunner.scala:125)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
    at java.lang.reflect.Method.invoke(Method.java:507)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
    at java.lang.reflect.Method.invoke(Method.java:507)
    at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
    at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:381)
    at py4j.Gateway.invoke(Gateway.java:259)
    at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
    at py4j.commands.CallCommand.execute(CallCommand.java:79)
    at py4j.GatewayConnection.run(GatewayConnection.java:209)
    at java.lang.Thread.run(Thread.java:785)
```

The notebook:

https://apsportal.ibm.com/analytics/notebooks/c94ee063-dab3-4440-9b90-125347ff80af/view?access_token=1c752de564c7c2df3fe924ce4122e71e658e73ff64160c80b667844527a8df88

**Comment by DTAIEB** [comment]
Good catch! The problem was that I was missing unicode support in transfer of python variable to scala. The issue has been fixed in v0.58. Please upgrade before testing.

BTW: I noticed in your notebook that you are calling ssc.awaitTermination(). This will cause the cell to never return. I don't think this is intended since there is no timeout code I could see. You might want to review this part :-)

**Comment by snowch** [comment]
Thanks for the quick turnaround @DTAIEB!
